### PR TITLE
Enabled OSGi metadata in MANIFST-MF for created JAR

### LINKED ISSUE #822: Add OSGi metadata
Do you mind adding some OSGi metadata to the project, so it can be used easily in an OSGi environment?
Like so:
https://github.com/cnoelle/Java-WebSocket/commit/76b3094ba9fb0c393ae4b9f48951ae57d330493b

The only effect on the resulting jar should be some additional entries in the MANIFEST file, no further dependencies introduced.

### PR DESCRIPTION
**Pull request by ViToni**
Changed `pom.xml` so that the OSGi metadata is created and included in the resulting JAR.

## Description
Added the `bnd-maven-plugin` and its configuration to the `pom.xml` so that resulting JAR will contain OSGi metadata.

## Related Issue
Fixes #822 .

## Motivation and Context
Enables usage of the library in OSGi without the requiring the user to wrap the library.

## How Has This Been Tested?
The library JAR still gets created.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
(The same failing tests as before the change are still failing afterwards...)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -26,18 +41,138 @@
...
+            <plugin>
+                <groupId>biz.aQute.bnd</groupId>
+                <artifactId>bnd-maven-plugin</artifactId>
+            </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-compiler-plugin</artifactId>
-                <version>3.7.0</version>
-                <configuration>
-                    <source>1.6</source>
```
Comment (marci4) [inline pom.xml]: Why did you remove this?

Comment (ViToni) [reply pom.xml]: @marci4 
The plugin configuration and version setting has been moved below `<pluginManagement>`.
This can be viewed as the default setup per plugin. Now one can just use the plugin without further configuration or the need to repeat the plugin version.

Additionally now all dependency / plugin versions are configured in a central place, see line 14-28. This has the benefit of having better control / overview of the versions used (eg. the javadoc plugin was used in two different versions before: 2.10.2 and 2.10.3).

Regarding your question:
The version of the compiler plugin can be found in line 22.
The configuration of the compiler plugin can be found in lines 95-103.

The source / target configuration could also be moved to `properties`:
`<maven.compiler.target>1.6</maven.compiler.target>`
`<maven.compiler.source>1.6</maven.compiler.source>`
but I tried to keep to the orginal style.
